sim/core/agent.go

```diff
@@ -125,7 +125,7 @@ func (actionID ActionID) ToProto() *proto.ActionID {
 	return protoID
 }
 
-func ProtoToActionID(protoID proto.ActionID) ActionID {
+func ProtoToActionID(protoID *proto.ActionID) ActionID {
```

Comment:
Why are we pointerizing everything? This was done to force the caller to handle the null case. If we're going to make these changes, the null handling logic needs to move with it.

Reply:
Pointerizing is because protobuf messages contain a mutex, and mustn't be copied: `copylocks: assignment copies lock value to X: github.com/wowsims/wotlk/sim/core/proto.XXX contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)`.

I've added an explicit nil-check here now, so it's obvious the caller must check this. Accessing `protoID.Tag` does cause a nil pointer dereference though, anyhow.